Source/BCfill.cpp

```diff
@@ -145,10 +189,94 @@ pc_bcfill_hyp(
   const int bcomp,
   const int scomp)
 {
-  const ProbParmDevice* lprobparm = PeleC::d_prob_parm_device;
+  ProbParmDevice* probparmDD =
+    PeleC::d_prob_parm_device; // probparm data for device
+  ProbParmDevice* probparmDH =
+    PeleC::h_prob_parm_device; // host copy of probparm data for device
+  ProbParmHost* probparmH = PeleC::prob_parm_host; // probparm data for host
+  constexpr int dim = AMREX_SPACEDIM;
+
+#ifdef PELEC_USE_TURBINFLOW
+  if (probparmH->do_turb) {
+
+    // Copy problem parameter structs to host
+    amrex::Gpu::copy(
+      amrex::Gpu::deviceToHost, probparmDD, probparmDD + 1, probparmDH);
+    for (int dir = 0; dir < dim; ++dir) {
+
+      auto bndryBoxLO = amrex::Box(amrex::adjCellLo(geom.Domain(), dir) & bx);
+      // if(bc[idir] == amrex::BCType::ext_dir)
+      if (bcr[1].lo()[dir] == EXT_DIR && bndryBoxLO.ok()) {
+        // Create box with ghost cells and set them to zero
+        amrex::IntVect growVect(amrex::IntVect::TheUnitVector());
+        int Grow = PeleC::numGrow();
+        for (int n = 0; n < dim; n++)
+          growVect[n] = Grow;
+        growVect[dir] = 0;
+        amrex::Box modDom = geom.Domain();
+        modDom.grow(growVect);
+        auto bndryBoxLO_ghost = amrex::Box(amrex::adjCellLo(modDom, dir) & bx);
+        data.setVal<amrex::RunOn::Host>(0.0, bndryBoxLO_ghost, UMX, dim);
+
+        add_turb(
+          bndryBoxLO, data, 0, geom, time, dir, amrex::Orientation::low,
+          probparmDH->tp);
+        probparmDH->turb_ok[dir] = true;
+      }
+
+      auto bndryBoxHI = amrex::Box(amrex::adjCellHi(geom.Domain(), dir) & bx);
+      if (bcr[1].hi()[dir] == EXT_DIR && bndryBoxHI.ok()) {
+        // Create box with ghost cells and set them to zero
+        amrex::IntVect growVect(amrex::IntVect::TheUnitVector());
+        int Grow = PeleC::numGrow();
+        for (int n = 0; n < dim; n++)
+          growVect[n] = Grow;
+        growVect[dir] = 0;
+        amrex::Box modDom = geom.Domain();
+        modDom.grow(growVect);
+        auto bndryBoxHI_ghost = amrex::Box(amrex::adjCellHi(modDom, dir) & bx);
+        data.setVal<amrex::RunOn::Host>(0.0, bndryBoxHI_ghost, UMX, dim);
+
+        add_turb(
+          bndryBoxHI, data, 0, geom, time, dir, amrex::Orientation::high,
+          probparmDH->tp);
+        probparmDH->turb_ok[dir + dim] = true;
+      }
+    }
+
+    // Copy problem parameter structs back to device
+    amrex::Gpu::copy(
+      amrex::Gpu::hostToDevice, probparmDH, probparmDH + 1, probparmDD);
+  }
+#endif
+
   amrex::GpuBndryFuncFab<PCHypFillExtDir> hyp_bndry_func(
-    PCHypFillExtDir{lprobparm});
+    PCHypFillExtDir{probparmDD});
   hyp_bndry_func(bx, data, dcomp, numcomp, geom, time, bcr, bcomp, scomp);
+
+#ifdef PELEC_USE_TURBINFLOW
```

Comment:
Why does this chunk of code exist? Seems like we are just setting some things to false? why is that?

Reply:
Probably not necessary, but is cleaning up after itself to return the state of the flags.  Yes, can remove

Reply:
We can indeed. I removed it from the LM implementation.